src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -596,7 +596,19 @@ def get_clouds(cli_ctx):
             # If profile isn't set, use latest
             setattr(c, 'profile', 'latest')
         if c.profile not in API_PROFILES:
-            raise CLIError('Profile {} does not exist or is not supported.'.format(c.profile))
+            if c.profile in (
+                "2017-03-09-profile",
+                "2018-03-01-hybrid",
+                "2019-03-01-hybrid",
+                "2020-09-01-hybrid",
+            ):
+                logger.error(
+                    "The azure stack profile '{}' has been deprecated and removed, using the 'latest' profile instead.\n"
+                    "To continue using Azure Stack please install the CLI `2.66.*` (LTS) version.".format(c.profile)
+                )
+                c.profile = 'latest'
```

Comment:
For azure stack users who already set the profile to azure stack profiles using old versions, this error messages will be displayed. Here we using the `latest` profile instead of raising the CLIError because the error will block all the commands including the `az cloud set --name xxx --profile latest` command which is used to fix the config.